python/surf/axi/_AxiStreamMonAxiL.py

```diff
@@ -0,0 +1,306 @@
+            pollInterval = 1,
+        )) 
+
+        self.add(pr.RemoteVariable(
+            name         = 'FrameRateMin',       
+            description  = "Min Frame Rate",
+            offset       = 0x14, 
+            bitSize      = 32, 
+            bitOffset    = 0, 
+            mode         = "RO",
+            base         = pr.Int, 
+            units        = 'Hz', 
+            pollInterval = 1,
+        ))
+        
+        addPair(
+            name         = 'Bandwidth',       
+            description  = "Current Bandwidth",
+            offset       = 0x18, 
+            bitSize      = 64, 
+            bitOffset    = 0, 
+            function     = self.convMbps,
+            units        = 'Mbps', 
+            pollInterval = 1,
+        )
+
+        addPair(
+            name         = 'BandwidthMax',       
+            description  = "Max Bandwidth",
+            offset       = 0x20, 
+            bitSize      = 64, 
+            bitOffset    = 0, 
+            function     = self.convMbps,
+            units        = 'Mbps', 
+            pollInterval = 1,
+        )
+
+        addPair(
+            name         = 'BandwidthMin',       
+            description  = "Min Bandwidth",
+            offset       = 0x28, 
+            bitSize      = 64, 
+            bitOffset    = 0, 
+            function     = self.convMbps,
+            units        = 'Mbps', 
+            pollInterval = 1,
+        )
+
+        self.add(pr.RemoteVariable(
+            name         = 'FrameSize',
+            description  = 'Current Frame Size. Note: Only valid for non-interleaved AXI stream frames',
+            offset       = 0x30,
+            bitSize      = 32,
+            bitOffset    = 0,
+            mode         = 'RO',
+            base         = pr.Int,
+            units        = 'Byte',
+            pollInterval = 1,
+        ))
+
+        self.add(pr.RemoteVariable(
+            name         = 'FrameSizeMax',
+            description  = 'Max Frame Size. Note: Only valid for non-interleaved AXI stream frames',
+            offset       = 0x34,
+            bitSize      = 32,
+            bitOffset    = 0,
+            mode         = 'RO',
+            base         = pr.Int,
+            units        = 'Byte',
+            pollInterval = 1,
+        ))
+
+        self.add(pr.RemoteVariable(
+            name         = 'FrameSizeMin',
+            description  = 'Min Frame Size. Note: Only valid for non-interleaved AXI stream frames',
+            offset       = 0x38,
+            bitSize      = 32,
+            bitOffset    = 0,
+            mode         = 'RO',
+            base         = pr.Int,
+            units        = 'Byte',
+            pollInterval = 1,
+        ))
+
+    @staticmethod
+    def convMbps(var):
+        return var.dependencies[0].value() * 8e-6        
+        
+class AxiStreamMonAxiL(pr.Device):
+    def __init__(self,       
+            name        = "AxiStreamMonitoring",
```

Comment:
This should be "AxiStreamMonAxiL".

Also, the Device `name` field gets auto populated with the class name, so in cases where you don't want it different (almost all cases), it's best to just omit it.

Reply:
Done. f0212af